server/src/kernels.cu

```diff
@@ -345,24 +505,80 @@ void diff::cuda::CUDACore::exec_core(uint8_t *frameData, uint8_t *showReadyNData
         kernel2_char<<<1, nThreadsToUse>>>(d_current, d_charsPx + idx * fullArea, eachThreadDoes, offset, 3 * charsSz.width, 3 * frameSz.width);
     }
 
+#ifdef NOISE_VISUALIZER
+#if NOISE_VISUALIZER == 4
+
+    // grayscale_kernel<<<1, nMaxThreads>>>(d_current, d_grayscale, maxAtTime);
+    // grayscale_kernel_v2<<<1, nMaxThreads>>>(d_current, d_grayscale, max4);
+    grayscale_kernel_v3<<<1, nMaxThreads>>>(d_current, d_grayscale, max4);
+    CUDA_CHECK(cudaMemcpyAsync(showReadyNData, d_grayscale, total, cudaMemcpyDeviceToHost));
+
+#elif NOISE_VISUALIZER == 5
+    int h_histogram[256] = {0};
+    uint8_t h_indexes[2] = {0};
+    int index_max = -1, index_sec_max = -1;
```

Comment:
Let me know If I can move them away from here

Reply:
I removed index_max and index_sec_max